x-pack/plugins/endpoint/server/routes/policy/handlers.test.ts

```diff
@@ -0,0 +1,138 @@
+    mockResponse = httpServerMock.createResponseFactory();
+    endpointAppContextService = new EndpointAppContextService();
+    mockAgentService = createMockAgentService();
+    endpointAppContextService.start({
+      indexPatternRetriever: createMockIndexPatternRetriever('metrics-endpoint-policy-*'),
+      agentService: mockAgentService,
+    });
+  });
+
+  afterEach(() => endpointAppContextService.stop());
+
+  it('should return the latest policy response for a host', async () => {
+    const response = createSearchResponse(new EndpointDocGenerator().generatePolicyResponse());
+    const hostPolicyResponseHandler = getHostPolicyResponseHandler({
+      logFactory: loggingServiceMock.create(),
+      service: endpointAppContextService,
+      config: () => Promise.resolve(EndpointConfigSchema.validate({})),
+    });
+
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+    const mockRequest = httpServerMock.createKibanaRequest({
+      params: { hostId: 'id' },
+    });
+
+    await hostPolicyResponseHandler(
+      createRouteHandlerContext(mockScopedClient, mockSavedObjectClient),
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockResponse.ok).toBeCalled();
+    const result = mockResponse.ok.mock.calls[0][0]?.body as GetHostPolicyResponse;
+    expect(result.policy_response.host.id).toEqual(response.hits.hits[0]._source.host.id);
+  });
+
+  it('should return not found when there is no response policy for host', async () => {
+    const hostPolicyResponseHandler = getHostPolicyResponseHandler({
+      logFactory: loggingServiceMock.create(),
+      service: endpointAppContextService,
+      config: () => Promise.resolve(EndpointConfigSchema.validate({})),
+    });
+
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+      Promise.resolve(createSearchResponse())
+    );
+
+    const mockRequest = httpServerMock.createKibanaRequest({
+      params: { hostId: 'id' },
+    });
+
+    await hostPolicyResponseHandler(
+      createRouteHandlerContext(mockScopedClient, mockSavedObjectClient),
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockResponse.notFound).toBeCalled();
+    const message = mockResponse.notFound.mock.calls[0][0]?.body;
+    expect(message).toEqual('Policy Response Not Found');
+  });
+});
+
+/**
+ * Create a SearchResponse with the hostPolicyResponse provided, else return an empty
+ * SearchResponse
+ * @param hostPolicyResponse
+ */
+function createSearchResponse(
+  hostPolicyResponse?: HostPolicyResponse
+): SearchResponse<HostPolicyResponse> {
+  return ({
+    took: 15,
+    timed_out: false,
+    _shards: {
+      total: 1,
+      successful: 1,
+      skipped: 0,
+      failed: 0,
+    },
+    hits: {
+      total: {
+        value: 5,
+        relation: 'eq',
+      },
+      max_score: null,
+      hits: hostPolicyResponse
+        ? [
+            {
+              _index: 'metrics-endpoint-policy-1',
```

Comment:
I think this needs to be `metrics-endpoint.policy-default-1`

Reply:
will change.